src/main/java/stobberi/components/Parser.java

```diff
@@ -0,0 +1,82 @@
+package stobberi.components;
+
+import stobberi.command.Command;
+import stobberi.command.DateCommand;
+import stobberi.command.DeadlineCommand;
+import stobberi.command.DeleteCommand;
+import stobberi.command.EventCommand;
+import stobberi.command.ExitCommand;
+import stobberi.command.FindCommand;
+import stobberi.command.ListCommand;
+import stobberi.command.MarkCommand;
+import stobberi.command.TodoCommand;
+import stobberi.command.UnmarkCommand;
+import stobberi.stobberiException.NoNumberStobberiException;
+import stobberi.stobberiException.NoSuchTaskStobberiException;
+import stobberi.stobberiException.StobberiException;
+
+/**
+ * This class is responsible for parsing user commands and creating the corresponding {@link Command} objects.
+ * The {@code parse} method interprets the user's command string and returns the appropriate command object based
+ * on the provided input. It throws specific exceptions for invalid or incomplete commands.
+ */
+public class Parser {
+
+    // Private constructor to prevent instantiation of the Parser class
+    private Parser() {}
+
+    /**
+     * Parses a user command and returns the corresponding {@link Command} object.
+     *
+     * @param fullCommand the full command string provided by the user
+     * @param taskList the task list to be used for command operations
+     * @return a {@link Command} object representing the user's command
+     * @throws StobberiException if the command is invalid or incomplete
+     * @throws NoNumberStobberiException if the command requires a number but it is missing
+     * @throws NoSuchTaskStobberiException if the command does not match any known commands
+     */
+    public static Command parse(String fullCommand, TaskList taskList) throws StobberiException {
+        String[] parts = fullCommand.split(" ");
+        String command = parts[0];
+        String restOfCommand = String.join(" ", java.util.Arrays.copyOfRange(parts, 1, parts.length));
+
+        switch (command) {
```

Comment:
Good job with the indentation of the switch statement. This is a common error for many so great job for avoiding it!

Reply:
Thank you!